src/otx/algo/diffusion/huggingface_model.py

```diff
@@ -0,0 +1,315 @@
+            target = noise
+        elif self.pipe.scheduler.config.prediction_type == "v_prediction":
+            target = self.pipe.scheduler.get_velocity(latents, noise, timesteps)
+        else:
+            error_msg = f"Invalid prediction type: {self.pipe.scheduler.config.prediction_type}"
+            raise ValueError(error_msg)
+
+        return {
+            "sample": noisy_latents,
+            "timestep": timesteps,
+            "encoder_hidden_states": encoder_hidden_states,
+            "target_noise": target,
+        }
+
+    def _customize_outputs(
+        self,
+        outputs: torch.Tensor,
+        inputs: DiffusionBatchDataEntity,
+    ) -> DiffusionBatchPredEntity | OTXBatchLossEntity:
+        return OTXBatchLossEntity(loss=outputs)
+
+    def on_fit_start(self) -> None:
+        """Called at the very beginning of fit.
+
+        If on DDP it is called on every process
+
+        """
+        self.pipe.to(self.device)
+        return super().on_fit_start()
+
+    def on_test_start(self) -> None:
+        """Called at the beginning of testing."""
+        self.pipe.to(self.device)
+        return super().on_test_start()
+
+    def on_predict_start(self) -> None:
+        """Called at the beginning of prediction."""
+        self.pipe.to(self.device)
+        return super().on_predict_start()
+
+    def validation_step(self, batch: DiffusionBatchDataEntity, batch_idx: int) -> None:
+        """Perform a single validation step on a batch of data from the validation set.
+
+        :param batch: A batch of data (a tuple) containing the input tensor of images and target
+            labels.
+        :param batch_idx: The index of the current batch.
+        """
+        # We use "latent" output type to overcome built-in NSFW filtering that blacks out some random images
+        images = self.pipe(batch.captions, output_type="latent").images
+        images = self.pipe.vae.decode(
+            images / self.pipe.vae.config.scaling_factor,
+            return_dict=False,
+        )[0]
+        images = self.pipe.image_processor.postprocess(
+            images,
+            output_type="pt",
+            do_denormalize=[True] * len(images),
+        )
+        self.metric.update(images, real=False)
+
+        # Save images
+        for logger in self.loggers:
+            if isinstance(logger, TensorBoardLogger):
+                for image, caption in zip(images, batch.captions):
+                    logger.experiment.add_image(f"val/images/{caption}", image.detach().cpu())
+
+    def test_step(self, batch: DiffusionBatchDataEntity, batch_idx: int) -> None:
+        """Perform a single test step on a batch of data from the test set.
+
+        :param batch: A batch of data (a tuple) containing the input tensor of images and target
+            labels.
+        :param batch_idx: The index of the current batch.
+        """
+        self.validation_step(batch, batch_idx)
```

Comment:
Same with above, because there is no `self.pipe` here.

Reply:
Fixed in [4a82d0d](https://github.com/openvinotoolkit/training_extensions/pull/3875/commits/4a82d0d2aa2f5b16803f10ac9f2d1725dd542682)